### PR TITLE
data en game opgesteld

### PR DESCRIPTION
**Pull request by broertje-cloud**
## 🎟️ Issue
 Linear ticket BEN-16 for example 

## ✨ Description
 Game,Question en answer aanngepast

## 🕵 Background
 Explain why this change is being made

## 📝 Resources
Add notes and documentation links where relevant

### CODE REVIEW

File: ExotischNederland/Models/Answer.cs [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
-    internal class Answer
+    public class Answer
     {
-        private string tablename = "Answer";
+        public int Id { get; set; }
+        public Question Question { get; set; }
+        public string Text { get; set; }
+        public bool Correct { get; set; }
+
+        private Answer(int id, Question question, string text, bool correct)
```
Comment (Underlyingglitch) [inline ExotischNederland/Models/Answer.cs:17]: Please use the constructor used in other files (with the Dictionary<string, object>

File: ExotischNederland/Models/Answer.cs [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+
+        // Static factory method to create a new Answer and save it to the database
+        public static Answer CreateAnswer(Question question, string text, bool correct)
+        {
+            Answer newAnswer = new Answer(0, question, text, correct);
+
+            // Save to database
+            SQLDAL db = new SQLDAL();
+            var values = new Dictionary<string, object>
+        {
```
Comment (Underlyingglitch) [inline ExotischNederland/Models/Answer.cs:33]: Careful with the line indentation here. The { and } should be 1 tab further for readability

File: ExotischNederland/Models/Game.cs [hunk 3]
```diff
@@ -1,13 +1,55 @@
...
+    public class Game
     {
-        private string tablename = "Game";
+        public int Id { get; set; }
+        public Route Route { get; set; }
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public List<Question> Questions { get; set; }
+
+        private Game(int id, Route route, string title, string description)
```
Comment (Underlyingglitch) [inline ExotischNederland/Models/Game.cs:18]: Use the other constructor here too

File: ExotischNederland/Models/Game.cs [hunk 4]
```diff
@@ -1,13 +1,55 @@
...
+        // Static factory method to create a new Game and save it to the database
+        public static Game CreateGame(int routeId, string title, string description)
+        {
+            Route route = new Route { Id = routeId };
+            Game newGame = new Game(0, route, title, description);
+
+            // Save to database
+            SQLDAL db = new SQLDAL();
+            var values = new Dictionary<string, object>
+        {
```
Comment (Underlyingglitch) [inline ExotischNederland/Models/Game.cs:36]: Watch the indentation here

File: ExotischNederland/Models/Route.cs
Comment (Underlyingglitch) [inline ExotischNederland/Models/Route.cs:1]: Why are these changes made? Does the Route need to be public? Id should never be settable from another class

**Review by Underlyingglitch — CHANGES_REQUESTED** [review]

**Review by broertje-cloud — COMMENTED** [review]
feedback aangepast

**Review by Underlyingglitch — CHANGES_REQUESTED** [review]
Unfortunately it still needs some work. I added some code refactors, so pull this branch first and check out what changed. For example: answers are added to the list on Question, but never stored/retrieved in the database. Same goes for the relationship Question - Game

**Review by broertje-cloud — COMMENTED** [review]
ok

**Pull request closed without merging**